src/RestClientVS/Margin/ResponseControl.xaml.cs

```diff
@@ -14,7 +17,43 @@ public async Task SetResponseTextAsync(RequestResult result)
         {
             if (result.Response != null)
             {
-                Control.Text = await result.Response.ToRawStringAsync();
+                var text = new StringBuilder();
+                switch (result.Response.Content.Headers.ContentType.MediaType)
+                {
+                    case "application/json":
```

Comment:
I wonder if it will catch all types if you check if the media type contained the string "json" instead.

Reply:
I can certainly do that - it should be relatively "safe" as we have the fallback for a failed parse operation - even if it captured something by mistake like "vnd.jsontario.data" that was actually text or some other type.